### ISSUE TITLE
# hadolint ignore=DL1000 not working

### ISSUE THREAD

**Issue by awhiriskey** [hadolint/hadolint#738]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
--mount=type=bind,target=/src,source=.,ro=false

ro=false causes hadolint to throw an error: DL1000: missing whitespace
 
When # hadolint ignore=DL1000 is added above the RUN command the error is still thrown. 

### Actual behavior

# hadolint ignore=DL1000
RUN --mount=type=bind,target=/src,source=.,ro=false pip3 install --no-cache -r /src/requirements-test.txt /src

ERROR: DL1000: missing whitespace

### Steps to reproduce the behavior

# hadolint ignore=DL1000
RUN --mount=type=bind,target=/src,source=.,ro=false pip3 install --no-cache -r /src/requirements-test.txt /src
EXPOSE 80

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

Haskell Dockerfile Linter v2.8.0-0-g398770f-dirty

```bash
./python/server/********.Dockerfile:21:46 missing whitespace
```

**Dockerfile** (if relevant)

```Dockerfile
#syntax=docker/dockerfile:1.0.0-experimental
ARG WORK_DIR=/app

FROM python:3.8-buster as base
ARG WORK_DIR
WORKDIR ${WORK_DIR}
RUN mkdir -p $HOME/.ssh \
    && ssh-keyscan -p 7999 RETACTED >> $HOME/.ssh/known_hosts
COPY requirements.txt ${WORK_DIR}
RUN --mount=type=ssh mkdir -p wheels && pip3 download -r requirements.txt -d wheels

FROM public.ecr.aws/lambda/python:3.8 as prod

ARG WORK_DIR
RUN mkdir wheels
COPY --from=base ${WORK_DIR}/wheels wheels
RUN  /var/lang/bin/python3.8 -m pip install --upgrade pip
RUN  pip3 install --no-cache-dir wheels/* && rm -rf wheels

# hadolint ignore=DL1000
RUN --mount=type=bind,target=/src,source=.,ro=false pip3 install --no-cache -r /src/requirements-test.txt /src
EXPOSE 80

COPY runserver.py /local/
COPY docker/entrypoint.sh /

ENTRYPOINT /entrypoint.sh

FROM prod as unit-test
ENTRYPOINT [ "pytest" ]

```

**Additional environment details (OS, stack version, etc.)**

**Comment by m-ildefons** [comment]
Hi @awhiriskey ,
DL1000 is a special error, because it indicates that Hadolint was not able to parse the Dockerfile or encountered some other error. Therefore it can not be ignored. This should probably be mentioned in the documentation somewhere.

The [syntax reference for buildkit](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypebind-the-default-mount-type) does not mention `ro` as valid argument for bind mounts, as they are read-only by default.
Though apparently either one of `ro=false`, `rw=true`, `rw`, `readonly=false`, `readwrite=true` or `readwrite` is accepted by buildkit.

You can make your Dockerfile work with Hadolint by replacing `ro=false` with just `rw` for now.
